src/http-client/fetch-client.ts

```diff
@@ -64,8 +69,88 @@ export class FetchClient implements HTTPClient {
     };
   }
 
+  /** {@inheritDoc HTTPStreamClient.stream} */
+  stream({
+    data,
+    headers: requestHeaders,
+    method,
+  }: HTTPStreamRequest): StreamAdapter {
+    const request = new Request(this.#streamURL, {
+      method,
+      headers: { ...requestHeaders, "Content-Type": "application/json" },
+      body: data,
+      keepalive: this.#keepalive,
+    });
+
+    const abortController = new AbortController();
+
+    const options = {
+      signal: abortController.signal,
+    };
+
+    async function* reader() {
+      const response = await fetch(request, options);
+      const body = response.body;
+      if (!body) {
+        throw new Error("Response body is undefined.");
+      }
+      const reader = body.getReader();
+
+      for await (const line of readLines(reader)) {
+        yield line;
+      }
+    }
+
+    return {
+      read: reader(),
+      close: () => {
+        abortController.abort();
+      },
+    };
+  }
+
   /** {@inheritDoc HTTPClient.close} */
   close() {
     // no actions at this time
   }
 }
+
+/**
+ * Get individual lines from the stream
+ *
+ * The stream may be broken into arbitrary chunks, but the events are delimited by a newline character.
+ *
+ * @param reader - The stream reader
+ */
+async function* readLines(reader: ReadableStreamDefaultReader<Uint8Array>) {
```

Comment:
A string in the response shouldn't ever include an actual `\n` character. I expect it would just be a literally the characters '\ and 'n'. But it would be good to check explicitly, and ultimately include in some test automation once we can use this in Node. 👍🏻